contracts/interfaces/IFees.sol

```diff
@@ -0,0 +1,62 @@
+pragma solidity 0.5.16;
+
+import "../IStakingContract.sol";
+import "./IContractRegistry.sol";
+
+/// @title Rewards contract interface
+interface IFees {
+    event FeesAssigned(address[] assignees, uint256 orbs_amount, uint256 bootstrap_amount);
+    event FeesAddedToBucket(uint256 bucketId, uint256 added, uint256 total);
+    event BootstrapAddedToPool(uint256 added, uint256 total);
+
+    /*
+     *   External methods
+     */
+
+    /// @dev Calculates and assigns validator fees and bootstrap fund for the time period since the last reward calculation
+    function assignFees() external returns (uint256);
+
+    /// @return Returns the currently unclaimed orbs token reward balance of the given address.
+    function getOrbsBalance(address addr) external view returns (uint256);
+
+    /// @return Returns the currently unclaimed bootstrap balance of the given address.
+    function getBootstrapBalance(address addr) external view returns (uint256);
+
+    /// @dev Transfer all of msg.sender's outstanding balance to their account
+    function withdrawFunds() external returns (uint256);
+
+    /// @return The timestamp of the last reward assignment.
+    function getLastFeesAssignment() external view returns (uint256);
+
+    /// @dev Transfers the given amount of bootstrap tokens form the sender to this contract and update the pool.
+    /// Assumes the tokens were approved for transfer
+    function topUpBootstrapPool(uint256 amount) external;
```

Comment:
Any reason to restrict the caller (e.g. taxing considerations)?

Reply:
i don't see a reason to restrict who can top up